apps/cmp.c

```diff
@@ -1024,11 +1024,15 @@ static int load_certs_also_pkcs12(const char *file, STACK_OF(X509) **certs,
     }
 
     for (i = 0; ret && i < sk_X509_num(*certs); i++) {
+        int cmp;
         cert = sk_X509_value(*certs, i);
-        if (OSSL_CMP_expired(X509_get0_notAfter(cert), vpm)) {
+        cmp = OSSL_CMP_cmp_timeframe(X509_get0_notBefore(cert),
+                                     X509_get0_notAfter (cert), vpm);
+        if (cmp != 0) {
             char *s = X509_NAME_oneline(X509_get_subject_name(cert), NULL, 0);
             BIO_printf(bio_err,
-                     "warning: certificate with subject '%s' has expired\n", s);
+                       "warning: certificate with subject '%s' %s\n", s,
+                       cmp > 0 ? "has expired" : "not yet valid");
             OPENSSL_free(s);
 #if 0
```

Comment:
#if 0 cleanup?

Reply:
Well, cleaning up `#if 0` (and `#if 1`) is unrelated to this PR, but nevertheless good to do.
So I've just done this for all of `cmp.c`.